test/ASTSpec.hs

```diff
@@ -125,24 +131,24 @@ spec = do
 31)) (Abs "b" (Meta (Infos {nthArg = Nothing, origin = Nothing}) (Area (Loc 79 3 24) (Loc 80 3 25)) (App (Meta (Infos {nthArg = Nothing, origin = Nothing}) (Area (Loc 79 3 24) (Loc 80 3 25)) (App (Meta (Infos {nthArg = Nothing, origin = Nothing}) (Area (Loc 81 3 26) (Loc 83 3 28)) (Var "+")) (Meta (Infos {nthArg = Nothing, origin = Nothing}) (Area (Loc 79 3 24) (Loc 80 3 25)) (App (Meta (Infos {nthArg = Nothing, origin = Nothing}) (Area (Loc 79 3 24) (Loc 80 3 25)) (App (Meta (Infos {nthArg = Nothing, origin = Nothing}) (Area (Loc 72 3 17) (Loc 75 3 20)) (Var "fn2")) (Meta (Infos {nthArg = Just 1, origin = Just (Var "fn2")}) (Area (Loc 76 3 21) (Loc 77 3 22)) (Var "a")))) (Meta (Infos {nthArg = Just 2, origin = Just (Var "fn2")}) (Area (Loc 79 3 24) (Loc 80 3 25)) (Var "b")))))) (Meta (Infos {nthArg = Nothing, origin = Nothing}) (Area (Loc 84 3 29) (Loc 85 3 30)) (Var "a")))))))))) (Meta (Infos {nthArg = Nothing, origin = Nothing}) (Area (Loc 38 2 7) (Loc 55 2 24)) (TRArr (Meta (Infos {nthArg = Nothing, origin = Nothing}) (Area (Loc 38 2 7) (Loc 41 2 10)) (TRSingle "Num")) (Meta (Infos {nthArg = Nothing, origin = Nothing}) (Area (Loc 45 2 14) (Loc 55 2 24)) (TRArr (Meta (Infos {nthArg = Nothing, origin = Nothing}) (Area (Loc 45 2 14) (Loc 48 2 17)) (TRSingle "Num")) (Meta (Infos {nthArg = Nothing, origin = Nothing}) (Area (Loc 52 2 21) (Loc 55 2 24)) (TRSingle "Num")))))))], aadts = [], apath = Just "/src/sourceA.mad"}),("/src/sourceB.mad",AST {aimports = [], aexps = [Meta (Infos {nthArg = Nothing, origin = Nothing}) (Area (Loc 0 1 1) (Loc 55 2 31)) (TypedExp (Meta (Infos {nthArg = Nothing, origin = Nothing}) (Area (Loc 0 1 1) (Loc 3 1 4)) (Export (Meta (Infos {nthArg = Nothing, origin = Nothing}) (Area (Loc 4 1 5) (Loc 6 1 7)) (Assignment "fn2" (Meta (Infos {nthArg = Nothing, origin = Nothing}) (Area (Loc 38 2 14) (Loc 55 2 31)) (Abs "a" (Meta (Infos {nthArg = Nothing, origin = Nothing}) (Area (Loc 38 2 14) (Loc 55 2 31)) (Abs "b" (Meta (Infos {nthArg = Nothing, origin = Nothing}) (Area (Loc 49 2 25) (Loc 50 2 26)) (App (Meta (Infos {nthArg = Nothing, origin = Nothing}) (Area (Loc 49 2 25) (Loc 50 2 26)) (App (Meta (Infos {nthArg = Nothing, origin = Nothing}) (Area (Loc 50 2 26) (Loc 52 2 28)) (Var "+")) (Meta (Infos {nthArg = Nothing, origin = Nothing}) (Area (Loc 49 2 25) (Loc 50 2 26)) (Var "a")))) (Meta (Infos {nthArg = Nothing, origin = Nothing}) (Area (Loc 53 2 29) (Loc 54 2 30)) (Var "b")))))))))))) (Meta (Infos {nthArg = Nothing, origin = Nothing}) (Area (Loc 7 1 8) (Loc 24 1 25)) (TRArr (Meta (Infos {nthArg = Nothing, origin = Nothing}) (Area (Loc 7 1 8) (Loc 10 1 11)) (TRSingle "Num")) (Meta (Infos {nthArg = Nothing, origin = Nothing}) (Area (Loc 14 1 15) (Loc 24 1 25)) (TRArr (Meta (Infos {nthArg = Nothing, origin = Nothing}) (Area (Loc 14 1 15) (Loc 17 1 18)) (TRSingle "Num")) (Meta (Infos {nthArg = Nothing, origin = Nothing}) (Area (Loc 21 1 22) (Loc 24 1 25)) (TRSingle "Num")))))))], aadts = [], apath = Just "/src/sourceB.mad"})])
```

Comment:
Not pretty, I'll have to use snapshots here as well as it's hard to build the correct expectation and editing the import inside the AST we build.